beacon-chain/sync/rpc_beacon_blocks_by_root.go

```diff
@@ -31,6 +33,17 @@ func (s *Service) sendRecentBeaconBlocksRequest(ctx context.Context, blockRoots
 		}
 		return nil
 	})
+
+	for _, blk := range blks {
+		blkRoot, err := blk.Block().HashTreeRoot()
```

Comment:
we can save on hashing by checking the block version here, if these are non deneb blocks, we simply skip it.

Reply:
[sendRecentBeaconBlocksRequest](https://github.com/prysmaticlabs/prysm/pull/12837/commits/83aeff3fee5bf6f1f8dae3c2db7709c4371df514)